language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/FixReturnTypeCodeAction.java

```diff
@@ -142,6 +147,10 @@ public List<CodeAction> getCodeActions(Diagnostic diagnostic,
                 }
             }
 
+            if (checkExprNodeFinder.containCheckExprNode()) {
```

Comment:
We can move the execution of the node visitor to the line before this line. No need to run it everytime (except in this else case)